src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -37,6 +40,8 @@ public enum Mode {
 
 	private Mode m_mode = Mode.BLUE_APRIL_TAGS; // By default we just using the limelight for localization
 
+	private BaseSwerveDrive m_SwerveDrive;
```

Comment:
Can you rename this to m_swerveDriveForSim? Or make it a pose supplier called m_simPoseSupplier?

Someone might miss that prepSimulation is the only spot where m_swerveDrive is set and try using it for non sim stuff

Reply:
Yeah sure. Kenny is working in this file and it will probably get overhauled soon, but hopefully this persists through the update.